OpenRA.Game/Manifest.cs

```diff
@@ -174,5 +179,29 @@ static string[] YamlList(Dictionary<string, MiniYaml> yaml, string key, bool par
 
 			return module;
 		}
+
+		static Dictionary<string, Manifest> LoadMods()
+		{
+			var basePath = Platform.ResolvePath(".", "mods");
+			var mods = Directory.GetDirectories(basePath)
+				.Select(x => x.Substring(basePath.Length + 1));
+
+			var ret = new Dictionary<string, Manifest>();
+			foreach (var mod in mods)
+			{
+				try
+				{
+					var manifest = new Manifest(mod);
+					ret.Add(mod, manifest);
+				}
+				catch (Exception ex)
+				{
+					Log.Write("debug", "An exception occured while trying to load mod {0}:", mod);
+					Log.Write("debug", ex.ToString());
```

Comment:
Unfortunately there is a ton of things that can go wrong when loading the Manifest, and then there's the possibility that `ret.Add(mod, manifest);` will throw (and that's a real possibility).
So I'm not sure I can come up with a more relevant exception message.